### DISCUSSION TITLE
Feature request:Analysis like Nvivo or MaxQda software do

### DISCUSSION THREAD

**Discussion by Rsose** [TriliumNext/Trilium#6161] [Plugin Ideas]
Hi Zadam！
Can you add some Anylisis features:
-Data Analysis
-Automatic generation of graphical analysis
-Frequency bands of word usage
-Generate and export professional models and charts
-Analyze how good or bad a thing is
Cos what I make notes now is on TRILIUM NOTES, but, for analysis, I still have to use NVIVO OR MaxQda to do analysis data.
It would be perfect if TRILIUM NOTES could provide similar analysis function
Mainly in TRILIUM NOTES export and then import to NVIVO process is very troublesome, hope that TRILIUM NOTES can make some analisis features in the future upgrades

[Nvivo Official Site](https://www.qsrinternational.com/nvivo-qualitative-data-analysis-software/home)
[MaxQda Features](https://www.maxqda.com/products/maxqda-standard/mixed-methods-functions)
[Nvivo videos](https://youtu.be/rnBBJDW4RhI)

Waiting a reply
Thanks

**Comment by Waxwood** [comment]
I also think this is a very good suggestion.  I have worked as a researcher for years and assist graduate students with their dissertation research and writing and have been searching for an affordable or low-cost qualitative data analysis application since 2007. A few open source QDA tools have appeared, but they are very limited. 

In fact, over the last 10 years, I've suggested this to developers of several note-taking programs to include QDA tools which would greatly expand the functionality of their software and enhance its appeal.

Please consider Rsose's suggestion:

Implement a QDA (Qualitative Data Analysis) feature- User can select portions sections of text and apply a keyword to the selected section. 

To achieve functionality as a QDA, in Trilium, a user would be able to select portions sections of text and apply a keyword to that entire section. Ideally, each keyword/tag would appear in a margin and indicate with highlighting or coloured bars each section of text that is coded with a particular keyword. By performing a search/analysis, all sections of text coded with particular keywords can be brought together in summary reports.

This would allow Trilium to function as a simple QDA (Qualitative data analysis software). Affordable QDA software essentially does not exist, and the open source initiatives underway are extraordinarily weak and years behind Trilium. 

Having QDA functionality would strengthen Trilium’s use as a data-sorting, cataloging and analysis tool. There are a great many esearchers and graduate students all over the world who want to use QDA software but cannot afford one of the “big names” and would be quite eager to adopt the use of Trilium as an open-source QDA.

**Comment by zadam** [comment]
Hi, to give some feedback on this issue - I have no experience with such software but some of the things you mention ("Analyze how good or bad a thing is") clearly seem out of scope of what I plan to do with Trilium.

That doesn't mean that none of it can be implemented. If you want some specific feature, then please create a separate issue about specific feature, provide rationale for it, ideally screenshot etc. That does not guarantee that I will work on it, but it still raises the chance from current 0 to something positive.

**Comment by Waxwood** [comment]
Hi Zadam, Rsose! Adding QDA (qualitative data analysis) functions to Trilium would be an incredible feature. I've followed many discussion forums in which researchers from around the world ask about low-cost or open source Qualitative Data Analysis software. Very few useful options exist. 

Regarding the definition of "analyse", we do not mean ascertain whether something is good or bad, but rather "To consider in detail and subject to an analysis in order to discover essential features or meaning". 

I've written a short description of these QDA features in the discussion here: https://github.com/zadam/trilium/discussions/1511
In the coming days, I'll put together some screenshots and explanations that should help clarify this type of QDA functionality.
